scripts/apps/authoring/macros/macros.js

```diff
@@ -191,6 +191,7 @@ function MacrosController($scope, macros, desks, autosave, $rootScope, storage,
             }
 
             if (isEditor3 || res.diff == null) {
+                $rootScope.macro_dirty = true;
```

Comment:
@petrjasek 
Any idea how to set dirty for the current item?

Reply:
what about `$scope.autosave()` ?

Reply:
Nemanja proposed first a more generic solution to use broadcast.
Please review it.

Reply:
imo using autosave like all other components is better

Reply:
OK, it is better autosave then. Please review again.